src/waveform/waveformwidgetfactory.cpp

```diff
@@ -142,13 +142,29 @@ WaveformWidgetFactory::WaveformWidgetFactory()
                     reinterpret_cast<const char*>(glFunctions->glGetString(GL_VENDOR))));
             QString rendererString = QString(QLatin1String(
                     reinterpret_cast<const char*>(glFunctions->glGetString(GL_RENDERER))));
-            qDebug() << versionString << vendorString << rendererString;
-
-            // note: the requested version has been set in WGLWidget's OpenGLWindow constructor
-            const int majorVersion = context->surface()->format().majorVersion();
-            const int minorVersion = context->surface()->format().minorVersion();
+            qDebug().noquote() << QLatin1String(
+                    "OpenGL driver version string \"%1\", vendor \"%2\", "
+                    "renderer \"%3\"")
+                                          .arg(versionString, vendorString, rendererString);
+
+            GLint majorVersion = -1;
+            glFunctions->glGetIntegerv(GL_MAJOR_VERSION, &majorVersion);
+            GLint minorVersion = -1;
+            glFunctions->glGetIntegerv(GL_MINOR_VERSION, &minorVersion);
+            if (majorVersion == -1 || minorVersion == -1) {
```

Comment:
If glGetIntegerv fails somehow I don't actually know what the value would be. I'm guessing it doesn't modify the passed in pointer, but I have no way to verify that.

Reply:
I guess the proper thing to do would be to check for `GL_INVALID_ENUM` errors? https://registry.khronos.org/OpenGL-Refpages/gl4/html/glGet.xhtml#errors
Also apparently you can rely on the `GL_VERSION` string to extract the version `GL_MAJOR/MINOR_VERSION` is not available...
https://www.khronos.org/opengl/wiki/OpenGL_Context#Context_information_queries

Reply:
done